WordPress/src/main/java/org/wordpress/android/ui/prefs/timezone/SiteSettingsTimezoneViewModel.kt

```diff
@@ -28,34 +33,55 @@ import java.time.zone.ZoneRulesException
 import java.util.Locale
 import javax.inject.Inject
 
-class SiteSettingsTimezoneViewModel @Inject constructor() : ViewModel() {
+class SiteSettingsTimezoneViewModel @Inject constructor(
+    private val resourceProvider: ResourceProvider
+) : ViewModel() {
     private val timezonesList = mutableListOf<TimezonesList>()
 
-    private val _showEmpty = MutableLiveData<Boolean>()
-    val showEmptyView: LiveData<Boolean> = _showEmpty
+    private val _showEmptyView = SingleLiveEvent<Boolean>()
+    val showEmptyView: LiveData<Boolean> = _showEmptyView
 
-    private val _showProgress = MutableLiveData<Boolean>()
-    val showProgressView: LiveData<Boolean> = _showProgress
+    private val _showProgressView = SingleLiveEvent<Boolean>()
+    val showProgressView: LiveData<Boolean> = _showProgressView
 
-    private val _dismiss = MutableLiveData<Unit>()
-    val dismissWithError: LiveData<Unit> = _dismiss
+    private val _dismissWithError = SingleLiveEvent<Unit>()
+    val dismissWithError: LiveData<Unit> = _dismissWithError
 
-    private val searchInput = MutableLiveData<String>()
-    val timezoneSearch: LiveData<List<TimezonesList>> = Transformations.switchMap(searchInput) { term ->
-        filterTimezones(term)
-    }
+    private val _dismissBottomSheet = SingleLiveEvent<Unit>()
+    val dismissBottomSheet: LiveData<Unit> = _dismissBottomSheet
 
     private val _suggestedTimezone = MutableLiveData<String>()
     val suggestedTimezone = _suggestedTimezone
 
+    private val _selectedTimezone = SingleLiveEvent<String>()
+    val selectedTimezone = _selectedTimezone
+
     private val _timezones = MutableLiveData<List<TimezonesList>>()
-    val timezones = _timezones
 
+    private val searchInput = MutableLiveData<String>()
+    private val timezoneSearch: LiveData<List<TimezonesList>> = Transformations.switchMap(searchInput) { term ->
+        filterTimezones(term)
+    }
+
+    val timezones = MediatorLiveData<List<TimezonesList>>().apply {
+        addSource(_timezones) {
+            value = it
+        }
+        addSource(timezoneSearch) {
+            value = it
+        }
+    }.distinctUntilChanged()
 
-    fun searchTimezones(query: CharSequence?) {
+
+    fun searchTimezones(query: CharSequence) {
         searchInput.value = query.toString()
     }
 
+    fun onTimezoneSelected(timezone: String) {
+        _selectedTimezone.value = timezone
+        _dismissBottomSheet.value = Unit
```

Comment:
TIP: There is a convenient `call()` or `asyncCall()` methods for situations like this :)

Reply:
Thanks for the tip, will use it